File: Programmers/HashMap/의상.java
import java.util.*;
class Solution {
    static HashMap<String, Integer> hm = new HashMap<>();
    public int solution(String[][] clothes) {
        int answer = 1;

        for (String[] tmp : clothes) {
            String type = tmp[1];
            // 1번 방식
            // if (hm.containsKey(type)) {
            //     hm.put(type, hm.get(type) + 1);
            // } else {
            //     hm.put(type, 1);
            // }

            // 2번 방식
            // hm.putIfAbsent(type, 0);
            // hm.put(type, hm.get(type) + 1);

            // 3번 방식
            hm.put(type, hm.getOrDefault(type, 0) + 1);
        }

        for (String key : hm.keySet()) {
            answer *= (hm.get(key) + 1);
        }

        return answer - 1; // 모든 옷을 입지 않는 경우 제외
    }
}

File: Programmers/HashMap/의상.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> hm;
    for (auto tmp : clothes) {
        string type = tmp[1];
        hm[type]++;
    }

    for (auto tmp : hm) {
        answer *= (tmp.second + 1); // 각 유형별로 선택하지 않는 경우의 수 1을 추가
    }
    return answer - 1; // 옷을 아예 입지 않는 경우 제외
}